util/util_test.go

```diff
@@ -905,3 +907,66 @@ func TestPathSliceSort(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyFs1(t *testing.T) {
+	src := afero.NewMemMapFs()
+	dst := afero.NewMemMapFs()
+
+	fooData := []byte("foo")
+	barData := []byte("bar")
+	bazData := []byte("baz")
+
+	src.MkdirAll("/foo/bar/baz/", 0700)
+	afero.WriteFile(src, "/foo/foo.txt", fooData, 0600)
+	afero.WriteFile(src, "/foo/bar/bar.txt", barData, 0600)
+	afero.WriteFile(src, "/foo/bar/baz/baz.txt", bazData, 0600)
+	CopyFs(src, dst, "", "", false)
+
+	contains, _ := afero.FileContainsBytes(dst, "/foo/foo.txt", fooData)
+	if !contains {
+		t.Errorf("Expected /foo/foo.txt to contain %s", fooData)
+	}
+
+	contains, _ = afero.FileContainsBytes(dst, "/foo/bar/bar.txt", barData)
+	if !contains {
+		t.Errorf("Expected /foo/bar/bar.txt to contain %s", barData)
+	}
+
+	contains, _ = afero.FileContainsBytes(dst, "/foo/bar/baz/baz.txt", bazData)
+	if !contains {
+		t.Errorf("Expected /foo/bar/baz/baz.txt to contain %s", bazData)
+	}
+}
+
+var dirInputs = []struct {
+	srcDirFull  string
+	srcCopyRoot string
+	dstCopyRoot string
+	dstExpected string
+	force       bool
+}{
+	{"/tmp/foo/bar/baz/", "/tmp/foo/", "/", "/foo/bar/baz", false},
+	{"/tmp/zoo/zar/zaz/", "/tmp/zoo/zar", "/start/dir", "/start/dir/zar/zaz", false},
+	{"/foo", "/foo", "/", "/foo", false},
+	{"/foo", "/foo", "/", "/foo", true},
+}
+
+func TestCopyFsDirTree(t *testing.T) {
+	for _, tt := range dirInputs {
+		src := afero.NewMemMapFs()
+		dst := afero.NewMemMapFs()
+
+		t.Run(tt.srcDirFull, func(t *testing.T) {
+			src.MkdirAll(tt.srcDirFull, 0700)
+			err := CopyFs(src, dst, tt.srcCopyRoot, tt.dstCopyRoot, tt.force)
+			if err != nil {
+				t.Errorf("Error copying source %s to dest %s", tt.srcCopyRoot, tt.dstCopyRoot)
```

Comment:
Errors are not full sentences, so they aren't capitalized. `golint` usually warns about this.